llvm/lib/CodeGen/ShrinkWrap.cpp

```diff
@@ -161,9 +161,9 @@ class ShrinkWrap : public MachineFunctionPass {
   /// Current MachineFunction.
   MachineFunction *MachineFunc = nullptr;
 
-  /// Is `true` for block numbers where we can guarantee no stack access
-  /// or computation of stack-relative addresses on any CFG path including
-  /// the block itself.
+  /// Is `true` for the block numbers where we cannot guarantee that there will
+  /// be no access to the stack or computation of stack-relative addresses on
+  /// any CFG path including the block itself.
```

Comment:
The comment is indeed out of date.
Is true => there may be a stack access or computation of ... (i.e., just drop the `no` from the original comment)
Is false => no stack access  or computation

Reply:
I agree that after my changes comment looks too complicated.
Nevertheless if I simply drop `no` from the original comment, we'll get
```
  /// Is `true` for block numbers where we can guarantee stack access
  /// or computation of stack-relative addresses on any CFG path including
  /// the block itself.
  ```
which means, what we are sure that there will be access to the stack or computation of stack-relative addresses. But really we only assume it.
May be it will be better to write
```
  /// Is `true` for block numbers where we assume possible stack access
  /// or computation of stack-relative addresses on any CFG path including
  /// the block itself.
  ```

Reply:
Yes, that’s I said “may”.
And call out the false case. Essentially we case have false positive, I.e., we are conservative to remain correct.

Feel free to rephrase!